src/hangar/dataloaders/torch_loader.py

```diff
@@ -0,0 +1,118 @@
+from collections import namedtuple
+try:
+    import torch
+    from torch.utils.data import Dataset
+except (ImportError, ModuleNotFoundError):
+    raise RuntimeError("Could not import torch. Try installing extra dependencies")
+
+
+class TorchDataset(Dataset):
+    """
+    TorchDataset inherits `torch.utils.data.Dataset` and accepts few convenient arguments
+    to wrap hangar datasets to be used in `torch.utils.data.DataLoaders`.
+
+    .. warning::
+
+        TorchDataset returns a namedtuple with `field_names` inferred from the dataset names
+        if not explicitly passed on init. Python named tuples has few restrictions on naming
+        the fields such it should not with an underscore etc. If your datatset names are not
+        good candidates for being `field_names` and if you haven't provided a good set of
+        `field_names` we'll rename those fields with namedtuples `rename=True`.
+
+    .. note::
+
+        From PyTorch 1.1 onwards, if Dataset returns namedtuple, DataLoader also returns
+        namedtuple
+
+    Parameters
+    ----------
+    hangar_datasets : `hangar.dataset.DatasetDataReader` or a Collection
+        A dataset object, a tuple of dataset object or a list of dataset objects`
+    keys : list or tuple
+        An array of sample names. If given only those samples will fetched from the dataset
+    index_range : slice
+        A python slice object which will be used to find the subset of dataset.
+        Argument `keys` takes priority over `index_range` i.e. if both are given, keys
+        will be used and `index_range` will be ignored
+    field_names : list or tuple of str
+        An array of field names used as the `field_names` for the returned namedtuple. If not
+        given, dataset names will be used as the field_names.
+
+    Examples
+    --------
+    >>> from hangar import Repository
+    >>> from torch.utils.data import DataLoader
+    >>> from hangar.dataloaders import TorchDataset
+    >>> repo = Repository('.')
+    >>> co = repo.checkout()
+    >>> dset = co.datasets['dummy_dset']
+    >>> torch_dset = TorchDataset(dset, index_range=slice(1, 100))
+    >>> loader = DataLoader(torch_dset, batch_size=16)
+    >>> for batch in loader:
+    ...     train_model(batch)
+
+    """
+
+    def __init__(self, hangar_datasets, keys=None, index_range=None, field_names=None):
+        if hasattr(hangar_datasets, 'get') and hasattr(hangar_datasets, 'name'):
```

Comment:
why not "if not isinstance(hangar_datasets, list):`?

Reply:
No good reason. Just felt safe with checking only two attributes of `Dataset` which I needed to use downstream